src/Platforms/Echo.Platforms.AsmResolver/Emulation/IFunctionMarshaller.cs

```diff
@@ -0,0 +1,26 @@
+using AsmResolver.DotNet;
+
+namespace Echo.Platforms.AsmResolver.Emulation;
+/// <summary>
+/// Provides methods for marshalling functions into pointers
+/// </summary>
+public interface IFunctionMarshaller
```

Comment:
Do we actually need a separate `IFunctionMarshaller` interface and `FunctionMarshaller` implementation class?

[ClrMockMemory](https://github.com/Washi1337/Echo/blob/master/src/Platforms/Echo.Platforms.AsmResolver/Emulation/Runtime/ClrMockMemory.cs) is already HLE'ing various internal structures of the CLR and mapping them to virtual addresses using `GenericMockMemory<T>`. I feel the proposed implementation is kind of reinventing the wheel for something that could simply be a `GenericMockMemory<IMethodDescriptor> MethodEntryPoints { get; }` in `ClrMockMemory`.